pkg/controller/command/verifiable/command.go

```diff
@@ -156,14 +156,21 @@ type kmsSigner struct {
 	bbs       bool
 }
 
-func newKMSSigner(keyManager kms.KeyManager, c ariescrypto.Crypto, creator string) (*kmsSigner, error) {
-	// creator will contain didID#keyID
-	idSplit := strings.Split(creator, "#")
-	if len(idSplit) != creatorParts {
-		return nil, fmt.Errorf("wrong id %s to resolve", idSplit)
+func getKID(opts *ProofOptions) string {
+	if opts.KID != "" {
+		return opts.KID
 	}
 
-	keyHandler, err := keyManager.Get(idSplit[1])
+	idSplit := strings.Split(opts.VerificationMethod, "#")
+	if len(idSplit) == creatorParts {
+		return idSplit[1]
+	}
+
+	return ""
+}
```

Comment:
is this logic still relevant? this is not a kms KID value

Reply:
I don't want to remove it because it will break compatibility (especially before release). Some services might depend on it.